src/commands/deploy.ts

```diff
@@ -3,28 +3,87 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-// tslint:disable-next-line:no-require-imports
-import WebSiteManagementClient = require('azure-arm-website');
+import * as fse from 'fs-extra';
+import * as path from 'path';
 import * as vscode from 'vscode';
+import { SiteWrapper } from 'vscode-azureappservice';
 import { AzureFunctionsExplorer } from '../AzureFunctionsExplorer';
-import { IUserInterface } from '../IUserInterface';
+import { NoPackagedJavaFunctionError } from '../errors';
+import { IUserInterface, Pick } from '../IUserInterface';
 import { localize } from '../localize';
 import { FunctionAppNode } from '../nodes/FunctionAppNode';
 import { getWebSiteClient } from '../nodes/SubscriptionNode';
+import { TemplateLanguage } from '../templates/Template';
+import { cpUtils } from '../utils/cpUtils';
+import { getProjectType } from '../utils/project';
 import * as workspaceUtil from '../utils/workspace';
 import { VSCodeUI } from '../VSCodeUI';
 
+/* tslint:disable:no-require-imports */
+import WebSiteManagementClient = require('azure-arm-website');
+import StringDictionary = require('azure-arm-website/lib/models/StringDictionary');
+
 export async function deploy(explorer: AzureFunctionsExplorer, outputChannel: vscode.OutputChannel, context?: FunctionAppNode | vscode.Uri, ui: IUserInterface = new VSCodeUI()): Promise<void> {
     const uri: vscode.Uri | undefined = context && context instanceof vscode.Uri ? context : undefined;
     let node: FunctionAppNode | undefined = context && context instanceof FunctionAppNode ? context : undefined;
 
-    const folderPath: string = uri ? uri.fsPath : await workspaceUtil.selectWorkspaceFolder(ui, localize('azFunc.selectZipDeployFolder', 'Select the folder to zip and deploy'));
+    let folderPath: string = uri ? uri.fsPath : await workspaceUtil.selectWorkspaceFolder(ui, localize('azFunc.selectZipDeployFolder', 'Select the folder to zip and deploy'));
 
     if (!node) {
         node = <FunctionAppNode>(await explorer.showNodePicker(FunctionAppNode.contextValue));
     }
 
     const client: WebSiteManagementClient = getWebSiteClient(node);
+    const siteWrapper: SiteWrapper = node.siteWrapper;
+    const languageType: string = await getProjectType(folderPath);
+    if (languageType === TemplateLanguage.Java) {
+        folderPath = await getJavaFolderPath(outputChannel, folderPath, ui);
+        await updateJavaFunctionAppSettings(outputChannel, client, siteWrapper);
+    }
+
+    await siteWrapper.deployZip(folderPath, client, outputChannel);
+}
+
+async function getJavaFolderPath(outputChannel: vscode.OutputChannel, basePath: string, ui: IUserInterface): Promise<string> {
```

Comment:
Theres should only ever be one package per function app project. You could use `project.properties.functionAppName` in `pom.xml` to determine the name of the package.

Reply:
Yes, that's a good idea. I Will confirm with the Maven plugin team to check if it is safe to do that. Let's file a issue for this, shall we?

Reply:
Here's what the outputDirectory is in pom.xml:
```
<outputDirectory>${project.build.directory}/azure-functions/${functionAppName}</outputDirectory>
```

You already hard-coded `azure-functions` into the path from your code, so I don't see why it would be a problem to also use functionAppName. Technically it is possible the user could change this, but I think the benefits for most users outweighs the negatives in this specific edge case.

Reply:
I'm okay with occasionally filing issues and addressing feedback after a PR is merged, but I don't like approving a PR before the issue is filed (otherwise it could be forgotten).

I want to merge code as soon as it's ready so I filed the issue for you this time (https://github.com/Microsoft/vscode-azurefunctions/issues/74). In the future, please just file the issue rather than asking "Let's file an issue for this, shall we?"